clean_air/util/file_converter.py

```diff
@@ -0,0 +1,171 @@
+                     'south': row[1].values[41],
+                     'east': row[1].values[43],
+                     'west': row[1].values[44],
+                     'chemicals': row[1].values[46].split(';'),
+                     'obs_level': row[1].values[19],
+                     'data_source': row[1].values[20],
+                     'time_range_start': row[1].values[37],
+                     'time_range_end': row[1].values[38],
+                     'lineage': row[1].values[50],
+                     'quality': row[1].values[51],
+                     'docs': row[1].values[22]}
+
+        form = pd.Series(data=form_data)
+        form_responses.append([form, r])
+
+    return form_responses
+
+
+def save_as_json(data_object, r, output_location):
+    """
+    Uses data held in pandas DataFrame to enter into form template and
+    save as JSON string.
+    """
+    # Pollutants(chemicals) must be organised into a useable structure before
+    # being entered into the new file:
+    chemicals = []
+    for chem in data_object.chemicals:
+        if chem != "":
+            chems = {'name': chem,
+                     'shortname': chem[chem.find("(") + 1:chem.find(")")],
+                     'chart': "url/to/chart/image.(png|jpg|svg|etc)"}
+            chemicals.append(chems)
+
+    new_file = {"pollutants": chemicals,
+                "environmentType": data_object.obs_level,
+                "dateRange": {"startDate": data_object.time_range_start,
+                              "endDate": data_object.time_range_end}}
+
+    # write the dictionary above (new_file) to a json with the addition of
+    # a unit to indicate the number of the metadata form response.
+    filename = "form_response" + str(r) + ".json"
+    with open(os.path.join(output_location, filename), 'w') as fp:
+        json.dump(new_file, fp, indent=2, cls=DateTimeEncoder)
+
+
+def save_as_yaml(data_object, r, output_location):
+    """
+    Uses data held in pandas DataFrame to enter into form template and
+    save as yaml.  Each data object entered here should be a single
+    response from the previous form input section.
+    """
+    # First extract the shortname only for chemical species:
+    chem_species = []
+    for chem in data_object.chemicals:
+        if chem != "":
+            chem_shortname = chem[chem.find("(") + 1:chem.find(")")]
+            chem_species.append(chem_shortname)
+
+    authors = {}
+    for i in range(1, 3):
+        firstname = data_object.get(f"firstname{i}")
+        surname = data_object.get(f"surname{i}")
+        if firstname and surname:
+            if i == 1:
+                authors.update({"firstname": firstname,
+                                "surname": surname})
+            elif (firstname and surname) is not type(str):
+                pass
+            else:
+                authors.update({f"firstname{i}": firstname,
+                                f"surname{i}": surname})
```

Comment:
I believe this had to be a dict in order to obtain both a key and a value for each entry.  The chemical species are collected as a list because they only need a simple list of entries, but this needs to be able to specify numbered firstname and surname entries if required.

Reply:
Here's what a list of 2 authors might look like:
```py
authors = [
    {"firstname": "John", "lastname": "Johnson"},
    {"firstname": "Jack", "lastname": "Jackson"},
]
```
and here's what this dict would look like with 2 authors:
```py
authors = {
    "firstname": "John",
    "lastname": "Johnson",
    "firstname1": "Jack",
    "lastname1": "Jackson",
}
```
Hopefully the list makes a bit more sense?

Reply:
I'd hope it would be able to handle a list of things directly. Then it can tell the difference between `authors[0]["firstname"]` and `authors[1]["firstname]`. But more importantly it's easier to identify that there are 2 authors in the first place, because `len(authors)` would be 2 regardless of how many fields make up one "author".

Reply:
ok, i'll change it.  i'm not very happy about it but you make a convincing arugment.